src/megatron/bridge/training/setup.py

```diff
@@ -191,8 +183,26 @@ def setup(
         cfg.model.register_pre_wrap_hook(peft_hook)
         print_rank_0("Registered PEFT pre-wrap hook")
 
+    # Megatron FSDP Config checks
+    if cfg.dist.use_megatron_fsdp:
+        assert cfg.dist.use_torch_fsdp2 is False, "use_megatron_fsdp and use_torch_fsdp2 cannot be True at the same time"
+
+        # Set Megatron FSDP Configs
+        cfg.model.use_megatron_fsdp = True
+        cfg.optimizer.use_megatron_fsdp = True
+        cfg.ddp.use_megatron_fsdp = True
+
+        if cfg.model.gradient_accumulation_fusion == True:
+            print_rank_0("Gradient accumulation fusion is not supported with Megatron FSDP, setting to False")
+            cfg.model.gradient_accumulation_fusion = False
+
+        if cfg.ddp.average_in_collective == True:
+            print_rank_0("average_in_collective is not supported with Megatron FSDP, setting to True")
+            cfg.ddp.average_in_collective = False
+
```

Comment:
why not have this run in `ConfigContainer.validate` instead? we should consider the config as frozen beyond this point: https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/685b95a5f2918cfee4a508f7ac12da8f206e29c4/src/megatron/bridge/training/setup.py#L111-L126

shouldn't this also set the `ckpt_format` to use pytorch DCP instead? is the mcore dist ckpt path fully compatible with megatron FSDP?

Reply:
Also I have moved the validation part to `ConfigContainer.validate()` now, thanks for the pointer